internal/model/rwfolder.go

```diff
@@ -1222,8 +1223,28 @@ func (p *rwFolder) performFinish(state *sharedPullerState) error {
 		}
 	}
 
-	// Set the correct timestamp on the new file
+	// Verify if a file will be overwritten and if we are up to date about its current mtime
 	t := time.Unix(state.file.Modified, 0)
+	existingInfo, existingErr := os.Lstat(state.realName)
+	if existingErr == nil {
+		p.model.fmut.RLock()
+		mtimeRepo, _ := p.model.folderMtimeRepos[p.folder]
+		p.model.fmut.RUnlock()
+		mtime := existingInfo.ModTime()
+		if mtimeRepo != nil {
+			mtime = mtimeRepo.GetMtime(state.file.Name, mtime)
+			// TODO should break here. If we need the mtime value from the mtimeRepo we are 'up to date' anyway
+		}
```

Comment:
If we need to check the time in the db, we now that the mtime is up to date anyway. A CRC check of the actual file would be much better here as we cannot rely on mtime on Android.